mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -68,6 +79,65 @@ class LogisticRegressionModel (
       case None => score
     }
   }
+
+  override def save(sc: SparkContext, path: String): Unit = {
+    val sqlContext = new SQLContext(sc)
+    import sqlContext._
+
+    // Create JSON metadata.
+    val metadata = LogisticRegressionModel.Metadata(
+      clazz = this.getClass.getName, version = Exportable.latestVersion)
+    val metadataRDD: DataFrame = sc.parallelize(Seq(metadata))
+    metadataRDD.toJSON.saveAsTextFile(path + "/metadata")
+    // Create Parquet data.
+    val data = LogisticRegressionModel.Data(weights, intercept, threshold)
+    val dataRDD: DataFrame = sc.parallelize(Seq(data))
+    dataRDD.saveAsParquetFile(path + "/data")
+  }
+}
+
+object LogisticRegressionModel extends Importable[LogisticRegressionModel] {
+
+  private case class Metadata(clazz: String, version: String)
+
+  private case class Data(weights: Vector, intercept: Double, threshold: Option[Double])
+
+  override def load(sc: SparkContext, path: String): LogisticRegressionModel = {
+    val sqlContext = new SQLContext(sc)
+    import sqlContext._
+
+    // Load JSON metadata.
+    val metadataRDD = sqlContext.jsonFile(path + "/metadata")
```

Comment:
That's not quite my question: I think the confusion is mixing "row" (line) in a text file vs "row" (or record) in an RDD.  How about we store the metadata in a single record in an RDD, but we print that RDD as multi-line JSON to a single text file?  It will be easier for humans to read and will be easy to load as a single record as well.